### PR TITLE
Create a minimum documentation page

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
This patch cleans up the Readme and creates a very minimal documentation page 
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

File: docs/advanced/developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+```bash
+IMAGE_REGISTRY=<my registry>
+make push
+```
+
+Alternatively, instead of specifying variables on the command line,
+you can edit the Makefile to permanently change parameter defaults
+like name of the image or namespace where the operator is deployed.
+
+## Manual deploy of the operator
```
Comment (marquiz) [inline docs/advanced/developer-guide.md:44]: s/deploy/deployment/ ?

File: docs/advanced/developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+make deploy
+```
+
+Then create a NodeFeatureDiscovery CR by running
+
+```bash
+kubectl apply -f config/samples/nfd.kubernetes.io_v1_nodefeaturediscovery.yaml
+```
+
+## Undeploy de operator
```
Comment (marquiz) [inline docs/advanced/developer-guide.md:59]: s/de/the/

File: docs/advanced/nodefeaturediscovery.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      #    - name: "my.combined.feature"
+      #      matchOn:
+      #        - pciId:
+      #            vendor: ["15b3"]
+      #            device: ["1014", "1017"]
+      #          loadedKMod : ["vendor_kmod1", "vendor_kmod2"]
+```
+
+For more information about how to setup the `WorkerConfig` stanza,
+see [worker config reference](https://kubernetes-sigs.github.io/node-feature-discovery/v0.8/advanced/worker-configuration-reference.html)
```
Comment (marquiz) [inline docs/advanced/nodefeaturediscovery.md:135]: Hmm, maybe the version of NFD that we're compatible with should be a parameter in the Jekyl config (`:config.yml`)?

File: docs/get-started/introduction.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+---
+title: "Quick start"
```
Comment (marquiz) [inline docs/get-started/introduction.md:2]: s/Quick start/Introduction/

File: docs/get-started/introduction.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+layout: default
+sort: 1
+---
+
+# Introduction
+
+The Node Feature Discovery Operator manages the detection
+of hardware features and configuration in an OpenShift
```
Comment (marquiz) [inline docs/get-started/introduction.md:10]: not only OpenShift 😄  Maybe Kubernetes?

File: docs/get-started/quick-start.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+---
+
+# Quick start
+
+Deploy the operator
+
+```bash
+IMAGE_TAG=k8s.gcr.io/nfd/node-feature-discovery-operator:v0.2.0
+make deploy
+```
```
Comment (marquiz) [inline docs/get-started/quick-start.md:14]: Maybe the first step should be cloning the repo?

File: docs/get-started/introduction.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+cluster by labeling the nodes with hardware-specific information.
+The Node Feature Discovery will label the host with
+node-specific attributes,
+like PCI cards, kernel, or OS version, and many more.
+
+Learn about the Node Feature Discovery Operator and
+how you can use it to expose node-level
+information by orchestrating the Node Feature Discovery,
+a Kubernetes add-on for detecting hardware features and system configuration!
```
Comment (marquiz) [inline docs/get-started/introduction.md:19]: This paragraph somehow feels disconnected/random 🤔  Learn where, how? Maybe extend the text/idea a bit, add links or smth?

File: docs/index.md [hunk 8]
```diff
@@ -4,6 +4,20 @@ layout: default
...
 
-# Introduction
+# Node Feature Discovery operator
 
-***COMING SOON.***
+The Node Feature Discovery operator manages detection
+of hardware features and configuration in a kubernetes cluster.
+The operator orchestrates all resources needed to run the
+[Node-Feature-Discovery](https://github.com/kubernetes-sigs/node-feature-discovery)
+DaemonSet
```
Comment (marquiz) [inline docs/index.md:13]: Maybe drop the Daemonset here? Just *The operator orchestrates all resources needed to run [Node-Feature-Discovery](https://github.com/kubernetes-sigs/node-feature-discovery)* ?

File: docs/index.md [hunk 9]
```diff
@@ -4,6 +4,20 @@ layout: default
...
+
+Continue to:
+
+- **[Quick start](get-started/index.md)** for quick step-by-step
+  instructions on how to get the NFD-Operator running on your cluster.
+
+- **[Advanced](advanced/index.md)** for Advanced topics and reference.
+
+- **[Contributing](contributing/index.md)** for information on how to reach the
+  community, project Governance and License info.
```
Comment (marquiz) [inline docs/index.md:23]: These links lead to empty-ish pages with no links or any "path forward". If we want to keep this index page the sub-index pages (get-started/index.md etc) should contain a toc, I think.

**Review by marquiz — CHANGES_REQUESTED** [review]
Nice work @ArangoGutierrez!

Some notes and comments when quickly skimming through the patch.

File: docs/advanced/developer-guide.md (hunk 2 above)
Comment (ArangoGutierrez) [reply docs/advanced/developer-guide.md:59]: my Spanish brain won the battle here ;P sorry

File: docs/index.md (hunk 9 above)
Comment (ArangoGutierrez) [reply docs/index.md:23]: I agree, will check how to do a toc in Jekyll ;P

File: docs/get-started/quick-start.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+Get the source code
+
+```bash
+git clone https://github.com/kubernetes-sigs/node-feature-discovery-operator
+```
+
+Deploy the operator
+
+```bash
+IMAGE_TAG=k8s.gcr.io/nfd/node-feature-discovery-operator:v0.2.0
```
Comment (marquiz) [inline docs/get-started/quick-start.md:18]: I'd suggest to parameterize the operator version in `docs/_config.yml`. Makes future updates of the version less error-prone as there's only one well-defined place where it needs to be changed.

File: docs/advanced/nodefeaturediscovery.md (hunk 3 above)
Comment (marquiz) [reply docs/advanced/nodefeaturediscovery.md:135]: What about this?

**Review by marquiz — CHANGES_REQUESTED** [review]
My only open comments are related to parameterizing the operator and operand versions in `docs/_config.yml`.

Also, we need some index.html in the site root. We can do e.g. redirection similar to nfd:
https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/docs/index.html

**Comment by ArangoGutierrez** [comment]
gotta say `verify-docs` is... very difficult to please

**Review by marquiz — CHANGES_REQUESTED** [review]
Woops 🙄 Now that I actually look at the config here (docs/_config.yml) I see that most of the things are already parameterized (`version`, `release`, `container_image`). Version is supposed to indicate the "release branch" (e.g. `v0.2`) where as release would match the full release tag (e.g. `v0.2.0`). The separate `container_image` variable is in place because staging (master) and releases use a different image registry. We'd actually only need to add `operand_release` there.

Also, the thinking behind the parameterization/versioning was that the master branch (what we're doing here) should actually instruct how to deploy the staging/master version of the operator. The reasoning behind is that otherwise the documentation and the version we instruct to deploy will (at some point) get out-of-sync and create confusion and/or stop to work. Getting user to deploy the latest release is handled by the documentation versioning (browsing to https://kubernetes-sigs.github.io/node-feature-discovery-operator will redirect the user to docs of the latest stable version).

Thus, my suggestion would be:
- only add `operand_release` parameter (in `docs/_config.yml`)
- use the existing `release`,  `version` and `container_image` elsewhere which will instruct the user to deploy the staging/devel version (for the master branch)
- when merged, backport this PR to `release-0.2` branch (should be quite trivial, prolly simply update `_config.yml`, plus, NodeFeatureDiscovery CR needs to be copied from v0.2.0) - this would create a "stable" version of the documentation in gh-pages where users would land by default

WDYT?

File: docs/advanced/nodefeaturediscovery.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+      #        - pciId:
+      #            vendor: ["15b3"]
+      #            device: ["1014", "1017"]
+      #          loadedKMod : ["vendor_kmod1", "vendor_kmod2"]
+```
+
+For more information about how to setup the `WorkerConfig` stanza,
+see [worker config reference](
+  https://kubernetes-sigs.github.io/node-feature-discovery/
+  {{ site.operand_version }}/advanced/worker-configuration-reference.html)
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply docs/advanced/nodefeaturediscovery.md:137]: grrr  thanks for the tip, yamlinter is killing me

**Review by marquiz — APPROVED** [review]
Thanks for your efforts @ArangoGutierrez! I think this is very good initial documentation 😄 👍 

/lgtm

**Pull request merged**